packages/ai-code-completion/src/common/code-completion-agent.ts

```diff
@@ -36,82 +36,89 @@ export class CodeCompletionAgentImpl implements CodeCompletionAgent {
         context: monaco.languages.InlineCompletionContext,
         token: monaco.CancellationToken
     ): Promise<monaco.languages.InlineCompletions | undefined> {
-        const languageModel =
-            await this.languageModelRegistry.selectLanguageModel({
-                agent: this.id,
-                ...this.languageModelRequirements[0],
+        const progress = await this.progressService.showProgress(
+            { text: 'Calculating AI code completion...', options: { location: 'window' } }
+        );
+        try {
+            const languageModel =
```

Comment:
nit: For the sake of easier extensibility, I wonder whether it makes sense to split some of the steps and the whole process out into a separate method that can then be overwritten. However, this is definitely not something that needs to be part of this PR if considered at all.

Reply:
We should do this, if we adapt the code completion to more LLMs.